tests/analyse.py

```diff
@@ -0,0 +1,49 @@
+from tests import queries
+
+
+def analyze_mining(deployment,last_layer, layers_per_epoch, layer_avg_size, total_pods):
+    # need to filter out blocks that have come from last layer
+    blockmap, layermap = queries.get_blocks_per_node_and_layer(deployment)
+    queries.print_node_stats(blockmap)
+    queries.print_layer_stat(layermap)
+
+    xl = [len(layermap[str(x)]) for x in range(layers_per_epoch) if str(x) in layermap]
+    print(str(xl))
+
+    first_epoch_blocks = sum(xl)
+
+    # count all blocks arrived in relevant layers
+    total_blocks = sum([len(layermap[str(x)]) for x in range(last_layer) if str(x) in layermap])
+
+    atxmap = queries.get_atx_per_node(deployment)
+    total_atxs = sum([len(atxmap[x]) for x in atxmap])
+
+    print("atx created " + str(total_atxs))
+    print("blocks created " + str(total_blocks) + " first epoch blocks " + str(first_epoch_blocks))
+
+    genesis_block_deviation = (total_pods - layer_avg_size) * 2 * layers_per_epoch
```

Comment:
not really needed anymore since your subtracting first epoch before assertion (L:33)